Source/DafnyLanguageServer/README.md

```diff
@@ -79,14 +79,32 @@ Options provided through the command line have higher priority than the options
 ### Plugins
 
 ```sh
-# Provides a path to assemblies that contain
-# 1) Either a class extending Microsoft.Dafny.Plugins.Configuration to receive plugin arguments
-#    and provide classes extending Microsoft.Dafny.Plugins.Rewriter
-# 2) Or, if no Configuration is defined, at least one Microsoft.Dafny.Plugins.Rewriter that it will load during resolution
-# Optionally, after a comma, provide an extra list of space-separated arguments.
+# Provides a path to assemblies and optional space-separated command-line arguments after a commo.
 # Easier to use via VSCode's Settings interface (no need to escape inner double quotes)
-# Default: "" (nothing extra is loaded)
+# Repeat with --dafny:plugins:0=... --dafny:plugins:1=... for multiple plugins.
 --dafny:plugins:0=example.dll
---dafny:plugins:1=example2.dll,oneArgument
-"--dafny:plugins:1=\"example2.dll,\\\"firstArgument with space\\\" secondArgument\""
+--dafny:plugins:0=example2.dll,oneArgument
+"--dafny:plugins:0=\"example2.dll,\\\"firstArgument with space\\\" secondArgument\""
 ```
+
+#### About plugins
```

Comment:
I think it would be good to state somewhere that all of the APIs are likely to change. Through the types of the API such as `Program`, we're exposing a lot of Dafny code that is likely to change.

Reply:
I just added an italicized paragraph at the beginning of this section that explains exactly this.
https://github.com/dafny-lang/dafny/blob/resolving-plugin-support/Source/DafnyLanguageServer/README.md#about-plugins
It should also address one concern of @robin-aws

Reply:
It does, and thanks for that!

Shouldn't this be be located somewhere more central though, since you can provide plugins on the command line as well? You can still have a section here about how to configure the language server to use plugins, but link to wherever we have the content below for the details about the plugin interface.

Reply:
The primary use case of plugins being to report errors for the language server for now, it currently makes sense to put it in its README.md.
There is a link from DafnyOptions.cs to this README.md
I think that, after implementing compilers on the plugins API, it will be time to migrate this documentation elsewhere, what do you think, @cpitclaudel ?